File: src/contests/cg/puzzle/dart/horse-racing-duals.dart
// 🎮 CodinGame Puzzle - horse-racing-duals
// https://www.codingame.com/training/easy/horse-racing-duals

import 'dart:io';

void main() {
  int n = int.parse(stdin.readLineSync()!.trim());
  List<int> s = [];
  for (int i = 0; i < n; i++) s.add(int.parse(stdin.readLineSync()!.trim()));
  s.sort();
  int minDiff = 1 << 62;
  for (int i = 1; i < n; i++) {
    int diff = s[i] - s[i - 1];
    if (diff < minDiff) minDiff = diff;
  }
  print(minDiff);
}

File: src/contests/cg/puzzle/ocaml/horse-racing-duals.ml
(* 🎮 CodinGame Puzzle - horse-racing-duals *)
(* https://www.codingame.com/training/easy/horse-racing-duals *)

let () =
  let n = int_of_string (String.trim (read_line ())) in
  let s = Array.make n 0 in
  for i = 0 to n - 1 do
    s.(i) <- int_of_string (String.trim (read_line ()))
  done;
  Array.sort compare s;
  let min_diff = ref max_int in
  for i = 1 to n - 1 do
    let diff = s.(i) - s.(i - 1) in
    if diff < !min_diff then min_diff := diff
  done;
  Printf.printf "%d\n" !min_diff
